src/test/java/com/commercetools/sync/internals/helpers/PriceCompositeIdTest.java

```diff
@@ -0,0 +1,230 @@
+    private static final String CASE_2 = "with currency and countryCode";
+    private static final String CASE_3 = "with currency and channel";
+    private static final String CASE_4 = "with currency, countryCode and channel";
+    private static final String CASE_5 = "with currency and customerGroup";
+    private static final String CASE_6 = "with currency, countryCode and customerGroup";
+    private static final String CASE_7 = "with currency, channel and customerGroup";
+    private static final String CASE_8 = "with currency, countryCode, channel and customerGroup";
+
+    private static final String CASE_9 = "with currency and validFrom";
+    private static final String CASE_10 = "with currency, countryCode and validFrom";
+    private static final String CASE_11 = "with currency, channel and validFrom";
+    private static final String CASE_12 = "with currency, countryCode, channel and validFrom";
+    private static final String CASE_13 = "with currency, customerGroup and validFrom";
+    private static final String CASE_14 = "with currency, countryCode, customerGroup and validFrom";
+    private static final String CASE_15 = "with currency, channel, customerGroup and validFrom";
+    private static final String CASE_16 = "with currency, countryCode, channel, customerGroup and validFrom";
+
+    private static final String CASE_17 = "with currency and validUntil";
+    private static final String CASE_18 = "with currency, countryCode and validUntil";
+    private static final String CASE_19 = "with currency, channel and validUntil";
+    private static final String CASE_20 = "with currency, countryCode, channel and validUntil";
+    private static final String CASE_21 = "with currency, customerGroup and validUntil";
+    private static final String CASE_22 = "with currency, countryCode, customerGroup and validUntil";
+    private static final String CASE_23 = "with currency, channel, customerGroup and validUntil";
+    private static final String CASE_24 = "with currency, countryCode, channel, customerGroup and validUntil";
+
+    private static final String CASE_25 = "with currency, validFrom and validUntil";
+    private static final String CASE_26 = "with currency, countryCode, validFrom and validUntil";
+    private static final String CASE_27 = "with currency, channel, validFrom and validUntil";
+    private static final String CASE_28 = "with currency, countryCode, channel, validFrom and validUntil";
+    private static final String CASE_29 = "with currency, customerGroup, validFrom and validUntil";
+    private static final String CASE_30 = "with currency, countryCode, customerGroup, validFrom and validUntil";
+    private static final String CASE_31 = "with currency, channel, customerGroup, validFrom and validUntil";
+    private static final String CASE_32 =
+        "with currency, countryCode, channel, customerGroup, validFrom and validUntil";
+
+    @ParameterizedTest(name = "[#ofPriceDraft]: {0}")
```

Comment:
Yes, I'm loving Junit5 till now. Still want to investigate Spock in the future. Both have their advantages and disadvantages I guess. For me the biggest disadvantage with spock is writing in groovy -> so implementation code will be different syntax than test code. 


Also, regarding the hard to read stuff, I think the tests in the link you sent are very trivial compared to actual tests in bigger projects like this lib, so I don't think in spock it will be That much easier ;)